src/template.ts

```diff
@@ -0,0 +1,301 @@
+						.onChange(value => {
+							this.config.bodyTemplate = value;
+						});
+					text.inputEl.rows = 6;
+				});
+
+			// Buttons
+			const buttonContainer = container.createDiv('modal-button-container');
+			buttonContainer.createEl('button', { cls: 'mod-cta', text: 'Continue' }, el => {
+				el.addEventListener('click', () => {
+					// Validate configuration
+					if (!this.config.titleTemplate.trim()) {
+						new Notice('Please provide a note title template.');
+						return;
+					}
+
+					resolve(this.config);
+				});
+			});
+
+			buttonContainer.createEl('button', { text: 'Cancel' }, el => {
+				el.addEventListener('click', () => {
+					resolve(null);
+				});
+			});
+		});
+	}
+}
+
+/**
+ * Applies a template string to a data object, replacing {{fieldName}} placeholders with values.
+ * 
+ * @param template - Template string with {{fieldName}} placeholders
+ * @param data - Object containing field values
+ * @returns Processed string with placeholders replaced
+ * 
+ * @example
+ * ```ts
+ * const result = applyTemplate('Hello {{name}}!', { name: 'World' });
+ * // Returns: "Hello World!"
+ * ```
+ */
+export function applyTemplate(template: string, data: Record<string, string>): string {
+	if (!template) return '';
+
+	return template.replace(/\{\{\s*([^}]+?)\s*\}\}/g, (match, fieldName) => {
+		const trimmedName = fieldName.trim();
+		return data[trimmedName] !== undefined ? data[trimmedName] : match;
+	});
+}
+
+/**
+ * Generates YAML frontmatter from data using property mappings.
+ * 
+ * @param data - Object containing field values
+ * @param propertyNames - Maps field IDs to property names
+ * @param propertyValues - Maps field IDs to property value templates
+ * @param convertToYAML - Function to convert string values to proper YAML format
+ * @returns YAML frontmatter string with --- delimiters, or empty string if no properties
+ * 
+ * @example
+ * ```ts
+ * const propertyNames = new Map([['title', 'title'], ['date', 'created']]);
+ * const propertyValues = new Map([['title', '{{title}}'], ['date', '{{date}}']]);
+ * const data = { title: 'My Note', date: '2025-10-01' };
+ * 
+ * const frontmatter = generateFrontmatter(data, propertyNames, propertyValues, convertToYAML);
+ * // Returns:
+ * // ---
+ * // title: My Note
+ * // created: 2025-10-01
+ * // ---
+ * ```
+ */
+export function generateFrontmatter(
+	data: Record<string, string>,
+	propertyNames: Map<string, string>,
+	propertyValues: Map<string, string>,
+	convertToYAML: (value: string) => string
+): string {
+	if (propertyNames.size === 0) return '';
+
+	const lines = ['---'];
+
+	for (const [fieldId, propertyName] of propertyNames) {
+		if (!propertyName) continue; // Skip if property name is empty
+
+		const valueTemplate = propertyValues.get(fieldId) || '';
+		if (!valueTemplate) continue; // Skip if property value is empty
+
+		// Apply the template to get the actual value
+		const value = applyTemplate(valueTemplate, data);
```

Comment:
We need to make sure that string values do not have newlines. Our app doesn't support that and will show the yaml frontmatter as invalid.